src/internal.rs

```diff
@@ -233,9 +237,13 @@ pub struct Rav1dContext {
     pub(crate) fc: *mut Rav1dFrameContext,
     pub(crate) n_fc: c_uint,
 
-    pub(crate) tc: *mut Rav1dTaskContext,
-    pub(crate) tc_shared: Vec<Rav1dContextTaskThread>,
+    /// Task context if we aren't using worker threads
+    pub(crate) main_tc: Option<Mutex<Box<Rav1dTaskContext>>>,
```

Comment:
Why do you need this `Box`?

Reply:
I wasn't sure about adding significant size to the `Rav1dContext` struct. On second thought, I don't think it matters here, I'll remove the Box.

Reply:
Ahhh.  I don't think `Rav1dContext` is ever really cloned or moved much.  Good consideration, though.

Reply:
And on third thought, I'm keeping it. The TaskContext is large and I don't want to pay extra memory in the threaded case where we don't use it.

Reply:
Okay, that's fine.  Could you leave a comment that that's why the `Box` is there?  Since it's not very clear otherwise.